toqito/channel_props/is_extremal.py

```diff
@@ -0,0 +1,64 @@
+"""Determines whether a quantum channel is extremal."""
+
+import numpy as np
+from numpy.linalg import matrix_rank
+
+from toqito.channel_ops.choi_to_kraus import choi_to_kraus
+
+
+def is_extremal(phi: np.ndarray | list[np.ndarray | list[np.ndarray]], tol: float = 1e-9) -> bool:
+    """Determine whether a quantum channel is extremal.
+
+    A channel with Kraus operators {A_i} is extremal if and only if
+    the set {A_i^† A_j : i,j=1,...,r} is linearly independent.
+
+    The channel can be provided as either:
+      - A Choi matrix (np.ndarray), which is converted to a flat list of Kraus operators.
+      - A flat list of Kraus operators ([np.ndarray, ...]).
+      - A nested list of Kraus operators ([[np.ndarray, ...], ...]), in which case the
+        list is flattened.
+
```

Comment:
You may also want to refer to other docstrings in the toqito project (again, using [this](https://github.com/vprusso/toqito/blob/f93f5168db7dddd053700b2e92a8badea4509cd9/toqito/channel_props/choi_rank.py#L9) as an arbitrary example) where we provide sections on examples, references, etc. We would want to include something similar here as well.